src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -656,15 +656,23 @@ void checkConstraintLanguageVersionAndRuntimeSupportForConversion(SyntaxNode syn
             }
             else
             {
-                if (collectionTypeKind == CollectionExpressionTypeKind.ListInterface ||
+                if ((collectionTypeKind is CollectionExpressionTypeKind.ListInterface or CollectionExpressionTypeKind.List) ||
                     node.HasSpreadElements(out _, out _))
                 {
                     // Verify the existence of the List<T> members that may be used in lowering, even
                     // though not all will be used for any particular collection expression. Checking all
                     // gives a consistent behavior, regardless of collection expression elements.
                     _ = GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_List_T__ctor, diagnostics, syntax: syntax);
                     _ = GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_List_T__ctorInt32, diagnostics, syntax: syntax);
-                    _ = GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_List_T__ToArray, diagnostics, syntax: syntax);
+
+                    if (collectionTypeKind is CollectionExpressionTypeKind.List)
```

Comment:
I didn't understand why we need Add but not ToArray when targeting to List and ToArray but not Add when targeting to list interface.

Reply:
For list interface, we will use `List<T>.Add` potentially. Updated.